router/core/factoryresolver.go

```diff
@@ -61,19 +63,22 @@ type DefaultFactoryResolver struct {
 
 func NewDefaultFactoryResolver(
 	ctx context.Context,
-	transportFactory ApiTransportFactory,
+	transportOptions *TransportOptions,
 	baseTransport http.RoundTripper,
 	log *zap.Logger,
 	enableSingleFlight bool,
 	enableNetPoll bool,
 	natsPubSubBySourceID map[string]pubsub_datasource.NatsPubSub,
 	kafkaPubSubBySourceID map[string]pubsub_datasource.KafkaPubSub,
 ) *DefaultFactoryResolver {
+	transportFactory := NewTransport(transportOptions)
 
 	defaultHttpClient := &http.Client{
+		Timeout:   transportOptions.SubgraphTransportOptions.RequestTimeout,
 		Transport: transportFactory.RoundTripper(enableSingleFlight, baseTransport),
 	}
 	streamingClient := &http.Client{
+		Timeout:   transportOptions.SubgraphTransportOptions.RequestTimeout,
```

Comment:
please remove it, streaming client could not have a timeout

Reply:
🤦